src/pybind/mgr/cephadm/module.py

```diff
@@ -2576,7 +2580,7 @@ def assign_hosts(self):
         if not self.spec.placement.label and not self.spec.placement.hosts and self.spec.placement.count:
             logger.info("Found num spec. Looking for labeled hosts.")
             # TODO: actually query for labels (self.daemon_type)
-            candidates = self.scheduler.place([x[0] for x in self.get_hosts_func()],
+            candidates = self.scheduler.place([x for x in self.get_hosts_func(None)],
```

Comment:
```suggestion
            candidates = self.scheduler.place([x for x in self.get_hosts_func()],
```

Reply:
I added None because I coudln't get mypy to shut up, even though the Callable signature shows the arg as Optional.  :man_shrugging:

Reply:
but fwiw i think it's likely this will go away (or change significantly) in the next pass on the scheduler